src/bin/pg_autoctl/env_utils.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * src/bin/pg_autoctl/env_utils.c
+ *   Utility functions for interacting with environment settings.
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "postgres_fe.h"
+
+#include "env_utils.h"
+#include "log.h"
+
+/*
+ * get_env_variable checks for the environment variable and copies
+ * its value into provided buffer if the buffer is not null.
+ *
+ * Function returns the length of the environment variable.
+ *
+ * Return values
+ * - GETENV_ERROR (-2)  : incorrect variable name is provided
+ * - GETENV_NOT_FOUND (-1) : variable is not found
+ * - GETENV_EMPTY (0)  : variable is found but no value is set
+ * - (1+) length of the environment variable
```

Comment:
That's too complex. Let's simplify the API with returning a bool and storing the other bits safely in a struct, like in the other comment I made about that.